### DISCUSSION TITLE
22 marzo 2022 - esercizio 2

### DISCUSSION THREAD

**Discussion by IonutCicio** [sapienzastudentsnetwork/basi-di-dati-1#9] [Esami]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/40034429/34e42265-5df1-4173-b698-4b1a53f54cd6)

**Comment by Warcophyr** [comment]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/7142a9e8-18c8-4dc2-b2b6-489e2ce65a24)
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/19f3fade-73a5-4eba-8724-89a999c7f809)

**Reply by salicius** [reply]
Ciao, ma a cosa serve far e il secondo ciclo se tanto è uguale al primo ? Non è scontato che non ha cambiamenti ?

**Reply by IonutCicio** [reply]
> Ciao, ma a cosa serve far e il secondo ciclo se tanto è uguale al primo ? Non è scontato che non ha cambiamenti ?

L'algoritmo per [verificare se una decomposizione ha un looseless join](https://curiousci.github.io/university/basi-di-dati-1/basi-di-dati-presentazione.html#43) esegue un'iterazione solo se:
- non ci sono righe con tutte 'a'
- nel turno precedente ci sono state modifiche

Quindi, se uno deve eseguire l'algoritmo pedissequamente, fa anche la seconda iterazione, dove vede che non ci sono cambiamenti (per capire se ha raggiunto il punto fisso l'algoritmo deve effettivamente verificare che $f(x_n) = x_n$... di fatto, se $f(x_{n - 1}) = x_n$, nulla ti assicura che poi $x_n$ è un punto fisso senza controllare l'iterazione successiva... tu a occhio lo riesci a vedere perché la tabella è piccola)

**Reply by salicius** [reply]
Capito, grazie mille!

**Comment by niccolo-fato** [comment]
Soluzione analoga a @Warcophyr:
![2024-01-13 09 27 13](https://github.com/sapienzastudentsnetwork/bdi2324/assets/73995334/49975237-5421-43f6-8717-4a51ac22cddd)
![2024-01-13 09 27 24](https://github.com/sapienzastudentsnetwork/bdi2324/assets/73995334/4e00105b-6d1a-490b-992b-9a54cedf28af)
